src/en/mangahen/src/eu/kanade/tachiyomi/extension/en/mangahen/MangaHen.kt

```diff
@@ -0,0 +1,172 @@
+
+                    is TextFilter -> {
+                        if (it.state.isNotEmpty()) {
+                            it.state.split(",").filter(String::isNotBlank).map { tag ->
+                                val trimmed = tag.trim().lowercase()
+                                if (trimmed.startsWith('-')) {
+                                    tagSearch(trimmed.removePrefix("-"))?.let { tagInfo ->
+                                        excludeTags.add(tagInfo)
+                                    }
+                                } else {
+                                    tagSearch(trimmed)?.let { tagInfo ->
+                                        includeTags.add(tagInfo)
+                                    }
+                                }
+                            }
+                        }
+                    }
+                    else -> {}
+                }
+            }
+
+            addQueryParameter("name", query)
+
+            addQueryParameter("search", "1")
+            if (includeTags.isNotEmpty()) addQueryParameter("include_tags", includeTags.joinToString())
+            if (excludeTags.isNotEmpty()) addQueryParameter("exclude_tags", excludeTags.joinToString())
+            if (page > 1) addQueryParameter("page", page.toString())
+        }.build()
+
+        return GET(url, headers)
+    }
+
+    override fun searchMangaParse(response: Response): MangasPage = popularMangaParse(response)
+
+    // Details
+
+    override fun mangaDetailsParse(response: Response): SManga {
+        val document = response.asJsoup()
+        return SManga.create().apply {
+            val authors = document.select("a[href*=/circles/]").eachText().joinToString()
+            val artists = document.select("a[href*=/authors/]").eachText().joinToString()
+            initialized = true
+            title = document.select("h1.font-semibold").text()
+            author = if (authors.isEmpty()) artists else authors
+            artist = artists
+            genre = document.select("a[href*=/tags/]").eachText().joinToString()
+            description = buildString {
+                append("Categories: ", document.select("a[href*=/categories/]").text(), "\n")
+                append("Parodies: ", document.select("a[href*=/parodies/]").text(), "\n")
+                append("Circles: ", document.select("a[href*=/circles/]").text(), "\n\n")
+                append(document.select("tr:contains(page)").text(), "\n")
+                append(document.select("tr:contains(view)").text(), "\n")
+            }
+            thumbnail_url = document.selectFirst("img[src*=thumbnail].w-96")?.absUrl("src")
+        }
+    }
+
+    // Chapters
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        return listOf(
+            SChapter.create().apply {
+                name = "Chapter"
+                setUrlWithoutDomain(response.request.url.toString())
+            },
+        )
+    }

```

Comment:
override fetchChapterList and return the chapter, since you are not using the response here